File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.py
class Solution(object):
    def maxDepth(self, root):
        if root is None:
            return 0
        
        queue = collections.deque([root])
        depth = 0

        while queue:
            depth += 1
            for i in range(len(queue)):
                cur_root = queue.popleft()
                if cur_root.left:
                    queue.append(cur_root.left)
                if cur_root.right:
                    queue.append(cur_root.right)

        return depth
        

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxDepth(TreeNode root) {
        return countDepth(root);
    }

    private int countDepth(TreeNode node) {
        if (node == null) return 0;

        int left = countDepth(node.left);
        int right = countDepth(node.right);

        return Math.max(left, right) + 1;
    }
}